plugin/trino-sqlserver/src/main/java/io/trino/plugin/sqlserver/SqlServerClient.java

```diff
@@ -384,6 +387,29 @@ protected String createTableSql(RemoteTableName remoteTableName, List<String> co
         }
     }
 
+    @Override
+    public Connection getConnection(ConnectorSession session, JdbcOutputTableHandle handle)
+            throws SQLException
+    {
+        return getConnectionWithSnapshotIsolation(() -> super.getConnection(session, handle));
+    }
+
+    @Override
+    public Connection getConnection(ConnectorSession session, JdbcSplit split)
+            throws SQLException
+    {
+        return getConnectionWithSnapshotIsolation(() -> super.getConnection(session, split));
+    }
+
+    private Connection getConnectionWithSnapshotIsolation(ConnectionSupplier supplier)
+            throws SQLException
+    {
+        Connection connection = supplier.get();
+        // SQL Server's READ COMMITTED + SNAPSHOT ISOLATION is equivalent to ordinary READ COMMITTED in Oracle, PostgreSQL.
+        connection.setTransactionIsolation(TRANSACTION_SNAPSHOT);
```

Comment:
if this fails, the connection should be closed to prevent a leak

Reply:
Why could it fail?